### PR TITLE
App Engine Standard should always use the uberjar

### LINKED ISSUE #21564: Google Cloud guide - build via Maven plugin should use uber-jar packaging
### Describe the bug

Currently, the [Building the application via the Google Cloud Maven plugin](https://quarkus.io/guides/deploying-to-google-cloud#building-the-application-via-the-google-cloud-maven-plugin) does not instruct the user to add `quarkus.package.type=uber-jar` and the value of `<artifact>` in the config of the `appengine-maven-plugin` implies that the standard packaging should work too. However, the deployment fails with:

```
java.nio.file.NoSuchFileException: /workspace/quarkus/quarkus-application.dat
        at sun.nio.fs.UnixException.translateToIOException (UnixException.java:92)
        at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:111)
        at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:116)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel (UnixFileSystemProvider.java:219)
        at java.nio.file.Files.newByteChannel (Files.java:371)
        at java.nio.file.Files.newByteChannel (Files.java:422)
        at java.nio.file.spi.FileSystemProvider.newInputStream (FileSystemProvider.java:420)
        at java.nio.file.Files.newInputStream (Files.java:156)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun (QuarkusEntryPoint.java:44)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main (QuarkusEntryPoint.java:28)
```
I'm not an expert here but the `target/appengine-staging` directory contains a very limited subset of the artifatcs from the `target/quarkus`.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #21564

### CODE REVIEW

**Review by mkouba — APPROVED** [review]

**Comment by asodja** [comment]
Sorry if this if it is somehow unrelated, but I saw that change: 
I was actually using AppEngine Standard Java 11 with a `fast-jar`. I used gcloud to deploy artifacts and I had to also specify [a custom entry point](https://cloud.google.com/appengine/docs/standard/java11/runtime#application_startup).

But it might be true that if you use some maven-plugin for deploy it won't work.

**Comment by loicmathieu** [comment]
@asodja yes, this is not the **only** way to do it but it's the simplier way that works in all cases.

Feel free to enhance the guide with fast-jar (aka default packaging) and a default entry point.

**Comment by loicmathieu** [comment]
@mkouba did you check the wording or should I ask someone to do it ? 
I don't know your level of english but mine deserve someone to check it ;)

File: docs/src/main/asciidoc/deploying-to-google-cloud.adoc [hunk 1]
```diff
@@ -67,17 +67,17 @@ runtime: java11
 
 This will create a default service for your App Engine application.
 
-Then, you can choose to build the application by yourself or letting `gcloud` or the Google Cloud Maven plugin build it for you.
-
-=== Building the application manually
-
-Set up your application to be packaged as an uber-jar via your `application.properties` file:
+App Engine Standard cannot understand Quarkus' specific packaging layout, therefore, you must set up your application to be packaged as an uber-jar via your `application.properties` file:
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-google-cloud.adoc:70]:
```suggestion
App Engine Standard does not support the default Quarkus' specific packaging layout, therefore, you must set up your application to be packaged as an uber-jar via your `application.properties` file:
```

File: docs/src/main/asciidoc/deploying-to-google-cloud.adoc [hunk 2]
```diff
@@ -67,17 +67,17 @@ runtime: java11
...
 
 [source, properties]
 ----
 quarkus.package.type=uber-jar
 ----
 
+Then, you can choose to build the application by yourself or letting `gcloud` or the Google Cloud Maven plugin build it for you.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-google-cloud.adoc:77]:
```suggestion
Then, you can choose to build the application manually or delegating that responsibility to `gcloud` or the Google Cloud Maven plugin.
```

**Comment by loicmathieu** [comment]
@geoand I applied your suggestions

**Review by geoand — APPROVED** [review]

**Pull request merged**